src/rdata/txt.rs

```diff
@@ -0,0 +1,88 @@
+use std::str;
+
+use Error;
+use super::{Parse, RData};
+
+pub (crate) const TYPE: isize = 16;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Record(pub String);
+
+impl<'a> Parse<'a> for Record {
+    fn parse(rdata: &'a [u8]) -> Result<RData<'a>, Error> {
+        let len = rdata.len();
+        if len < 1 {
+            return Err(Error::WrongRdataLength);
+        }
+        let mut ret_string = String::new();
+        let mut pos = 0;
+        while pos < len {
+            let rdlen = rdata[pos] as usize;
+            pos += 1;
+            if len < rdlen + pos {
+                return Err(Error::WrongRdataLength);
+            }
+            match str::from_utf8(&rdata[pos..(pos+rdlen)]) {
+                Ok(val) => ret_string.push_str(val),
+                Err(e) => return Err(Error::TxtDataIsNotUTF8(e)),
+            }
+            pos += rdlen;
+        }
+        Ok(RData::TXT(Record(ret_string)))
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    use {Packet, Header};
+    use Opcode::*;
+    use ResponseCode::NoError;
+    use QueryType as QT;
+    use QueryClass as QC;
+    use Class as C;
+    use RData;
+
+    #[test]
+    fn parse_response_multiple_strings() {
+        let response = b"\x06%\x81\x80\x00\x01\x00\x01\x00\x00\x00\x00\
```

Comment:
Is this a real data chunk? I get single rdata record for facebook.

I used to believe concatenation of multiple records work across multiple RDATAs, not as chunks in a single rdata field, am I wrong?

Reply:
I simply moved this test from `parser.rs` as part of refactoring. Perhaps @rigdent can help answer this?

Reply:
I don't think it's a real data chunk. I based it off a real data chunk but changed names/stuff.

Reply:
Probably I've overlooked it in #20. As far as I understand there can't be multiple chunks in a single RDATA field.

There can be multiple ResourceRecords though, and when there are multiple the app should consider them as a single concatenated resource record.

Since generally, we don't try to join answers we should return a borrowed slice and remove all the concatenation code as far as I undertand. This is not a task of this PR though.